packages/compass-databases-navigation/src/collection-item.tsx

```diff
@@ -5,18 +5,20 @@ import {
   spacing,
   css,
   cx,
+  ItemActionControls,
+  SmallIcon,
 } from '@mongodb-js/compass-components';
+import type { ItemAction } from '@mongodb-js/compass-components';
 import { COLLECTION_ROW_HEIGHT } from './constants';
-import { ActionControls } from './item-action-controls';
-import type { NamespaceAction } from './item-action-controls';
 import { ItemContainer, ItemLabel } from './tree-item';
 import type {
   VirtualListItemProps,
   TreeItemProps,
   NamespaceItemProps,
 } from './tree-item';
-import { SmallIcon } from './icon-button';
-import type { Actions } from './constants';
+import { Actions } from './constants';
+
+const DBActionControls = ItemActionControls<Actions>;
```

Comment:
@gribnoysup Is this the right way to make a specialised version of that function component to use in jsx below? Feels wrong to me, but can't really explain why.

Reply:
I think you might not even need to do this as TS should be able to infer the type from the actions that you are passing to the component. But also yes, support for this partial application was actually added not that long ago to typescript and that exactly is the use-case for something like this

Reply:
oh I didn't even realise you can do this:

```
      <ItemActionControls<Actions>
        className={collectionActions}
        onAction={onAction}
        isActive={isActive}
        isHovered={isHovered}
        actions={actions}
        shouldCollapseActionsToMenu
      ></ItemActionControls>
```

Reply:
ok it isn't necessary, then. Will remove.